File: src/me/Qball/Wild/Utils/GetHighestNether.java
package me.Qball.Wild.Utils;

import org.bukkit.Location;
import org.bukkit.entity.Player;

public class GetHighestNether {

    public int getSolidBlock(int tempX, int tempZ, Player target) {
        for (int y = 124; y > 2; y--) {
            if (target.getWorld().getBlockAt(tempX, y, tempZ).isEmpty()) {
                Location loc = new Location(target.getWorld(), tempX, y, tempZ, 0.0F, 0.0F);
                if (target.getWorld().getBlockAt(tempX, loc.getBlockY() - 2, tempZ).isEmpty()
                        && !target.getWorld().getBlockAt(tempX, loc.getBlockY() - 4, tempZ).isEmpty()
                        && !target.getWorld().getBlockAt(tempX, loc.getBlockY() - 4, tempZ).isLiquid()) {
                    return y-3;
                }

            }
        }
        return 10;
    }


}

File: src/main/kotlin/me/qball/wild/util/GetHighestNether.kt
package me.qball.wild.util

import me.qball.wild.Wild
import org.bukkit.entity.Player

object GetHighestNether {
    private val plugin = Wild.instance
    private val randomLoc = GetRandomLocation()

    fun getSolidBlock(x: Int, z: Int, target: Player) : Int {
        var y = 0
        for(Y in 0..128) {
            y = Y
            if(target.world.getBlockAt(x, y, z).isEmpty && target.world.getBlockAt(x, y, z).isLiquid) {
                y += 2
                break
            }
        }
        return y
    }
}


